h/semTable.h

```diff
@@ -0,0 +1,7 @@
+  /* declare the table that maps datasets to semaphores */
+  struct sem_table_type {
```

Comment:
I don't think this header should exist in the first place. But if we had to have it, it must have a some sort of a include guard.